src/Microsoft.AspNet.Razor/Tokenizer/Tokenizer.cs

```diff
@@ -31,6 +35,12 @@ protected Tokenizer(ITextDocument source)
             StartSymbol();
         }
 
+        protected abstract int StartState { get; }
+
+        protected int? CurrentState { get; set; }
```

Comment:
Do we see allocations for this bit a lot? Could have a separate flag to showcase the tokenizer is done rather than rely on this being `null`.